classes/publication/Repository.php

```diff
@@ -456,6 +467,93 @@ public function makeThumbnail($filePath, $thumbFileName, $maxWidth, $maxHeight)
         }
     }
 
+    /**
+     * Remove DOI IDs from a previous version of a publication.
+     * Used in publication versioning when not versioning DOIs.
+     *
+     * @param Publication $oldPublication Publication with DOIs to be removed
+     * @param Publication $newPublication Publication that should receive copied DOIs
+     */
+    public function clearOldDois(Context $context, Publication $oldPublication, Publication $newPublication): void
+    {
+        parent::clearOldDois($context, $oldPublication, $newPublication);
+
+        // Publication Formats
+        $oldPubFormats = $oldPublication->getData('publicationFormats');
+        $newPubFormats = collect($newPublication->getData('publicationFormats'));
+
+        $representationDao = Application::getRepresentationDAO();
+
+        foreach ($oldPubFormats as $oldPubFormat) {
+            $newPubFormat = $newPubFormats->first(
+                fn ($pubFormat) => $pubFormat->getData('doiId') === $oldPubFormat->getData('doiId')
+            );
+            $doiId = $oldPubFormat->getData('doiId');
+
+            // Only remove old pub format DOI reference if a corresponding new one exists
+            if ($newPubFormat !== null && $doiId !== null) {
+                $oldPubFormat->setData('doiId', null);
+
+                $representationDao->updateObject($oldPubFormat);
+            }
+
+            // Submission files
+            // Only remove old DOI reference if a new publication format exists to include submission files
+            if ($newPubFormat !== null) {
```

Comment:
I think this would still be okay in that case. This is checking if there is a publication format at all, not checking if there are DOIs assigned to or enabled for it. As far as I know, these submission files can only be assigned in the context of a publication format. I've updated the comment to be more specific as well. Let me know what you think.